clientconn.go

```diff
@@ -196,7 +195,7 @@ func WithTimeout(d time.Duration) DialOption {
 }
 
 // WithDialer returns a DialOption that specifies a function to use for dialing network addresses.
-func WithDialer(f func(addr string, timeout time.Duration) (net.Conn, error)) DialOption {
+func WithDialer(f transport.DialerFn) DialOption {
```

Comment:
Yeah, that's fine. This was only done to reduce code duplication inside the grpc package. In other words, if you move `transport` to `internal`, all my changes will still work.